components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -89,4 +95,36 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
         }
         return null;
     }
+
+    /**
+     *Get federated authentication session mapping info during the federated idp init logout
+     *
+     * @param fedIdpSessionId
+     * @return A FederatedUserSession containing federated authentication session details
+     * @throws SessionManagementServerException
+     */
+    public FederatedUserSession getFederatedAuthSessionDetails(String fedIdpSessionId)
+            throws SessionManagementServerException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement preparedStatement =
+                     connection.prepareStatement(SQLQueries.SQL_GET_FEDERATED_AUTH_SESSION_INFO_BY_SESSION_ID)) {
+            preparedStatement.setString(1, fedIdpSessionId);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                FederatedUserSession federatedUserSession = new FederatedUserSession();
+                if (resultSet.next()) {
+                    federatedUserSession.setIdPSessionId(fedIdpSessionId);
+                    federatedUserSession.setSessionId(resultSet.getString("SESSION_ID"));
```

Comment:
Let's use constants for column names.

Also can you take a look at UserSession getSession(String sessionId) and see if we can follow the same approach with JdbcTemplates

Reply:
Implemented with the jdbc templates in 4ac6106